crates/subspace-service/src/sync_from_dsn.rs

```diff
@@ -280,6 +276,10 @@ where
             initial_sync_mode.take().unwrap_or(prev_sync_mode),
             Ordering::Release,
         );
+
+        while notifications.try_next().is_ok() {
```

Comment:
Is this a best effort fix? (i.e) if some notification is delayed (comes after this loop executes), it will be missed here and gets processed in next iteration anyways

Reply:
I think this is a proper fix. What I meant is that nodes that are not synced and not importing blocks for a while will re-sync from DSN after some time.

There is nothing wrong with skipping or repeating DSN sync, this change just provides a better user experience.

Reply:
ok. Looks like the reason is not used, may be it can be used to avoid redundant operations

Reply:
Reason is primarily for logging purposes